src/routes/lend.js

```diff
@@ -73,12 +74,27 @@ function LendPage() {
         }
     }, [loadOpensea, loadFakeNFT]);
 
+    const fetchAvailableListings = useCallback(() => {
+        const contract = KasuContract();
+        const filter = { address: contract.address,
+                         topics: [ethers.utils.id("ListNFT(uint256)")] };
+
+        LoginService.getInstance().provider.on(filter, event => {
+            const tokenID = Number(event.data);
+            nftsListedForLendingRef.current.add(tokenID);
+            setNFTsInUserWallet(nfts => {
+                return nfts.filter(obj => !nftsListedForLendingRef.current.has(obj.tokenID.toNumber()));
```

Comment:
I think you need to check both the tokenAddress and tokenID for the token to be unique